### PR TITLE
Feat: Added feature Export Visits to download the csv of Visits and Views

### LINKED ISSUE #786: [🕹️] Export document views to CSV
### What side quest or challenge are you solving?

Add an export function to download the document views to CSV by expanding the "..." button on the document detail page and select "Export visits".

It should trigger a direct download of a CSV file with the following content related to the visitor and visit:

- Viewed at
- Name (if available)
- Email (if available)
- Link Name
- Total Visit Duration
- % (Visit) Completion
- Document version
- Downloaded at
- Verified (if available)

The download file should be named after the document name and current timestamp e.g. `presentation-pdf_visits_2024-10-12_14-30.csv` or whatever makes more sense :)

### Points

750 Points

### Provide proof that you've completed the task

Open a PR with working code and explanation what you did. As well as a quick video showing the solution

### PR DESCRIPTION
**Pull request by aryany9**
close #786

This PR adds the Export functionality to download the Visits & Views counts in an csv file on "Export Visits" button click.

## Describe your changes
I have added an API to fetch the Views and Visits as per the document id and provide the JSON response in return.
image

Based on the Response, I am creating the CSV blob and downloading it on button click.
image

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] I have added quick video solution as requested below. 

https://github.com/user-attachments/assets/eee3d12f-0164-4fad-8583-0214de0be8ea

### CODE REVIEW

**Comment by aryany9** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by aryany9** [comment]
recheck

**Comment by aryany9** [comment]
@mfts  Please attach the oss.gg labels as well. They were not assigned by the bot.

File: package.json [hunk 1]
```diff
@@ -84,6 +84,7 @@
     "handlebars": "^4.7.8",
     "input-otp": "^1.2.4",
     "js-cookie": "^3.0.5",
+    "json2csv": "^6.0.0-alpha.2",
```
Comment (mfts) [inline package.json]: @aryany9 this package is abandoned. please remove it and change the approach
https://www.npmjs.com/package/json2csv

**Review by mfts — CHANGES_REQUESTED** [review]
@aryany9 the package you used is abandoned! please make sure you review packages before adding anything external. 

please change the approach to do it without an external library. 
for example, something like this: https://dev.to/graciesharma/implementing-csv-data-export-in-react-without-external-libraries-3030

**Comment by aryany9** [comment]
Hi @mfts, Thankyou for pointing out the abandoned package. It has been removed and the approach has been changed as well. I have tested it with the new approach it is working as expected. Do let me know if you require any more changes. Thankyou.

**Review by mfts — APPROVED** [review]
Looks amazing! Thanks so much for following the request and making it native without an external parser :)

**Pull request merged**